core/base64.js

```diff
@@ -13,33 +13,30 @@ export default {
 
     encode: function (data) {
         "use strict";
-        var result = '';
-        var toBase64Table = this.toBase64Table;
-        var length = data.length;
-        var lengthpad = (length % 3);
+        let result = '';
+        const length = data.length;
+        const lengthpad = (length % 3);
         // Convert every three bytes to 4 ascii characters.
 
-        for (var i = 0; i < (length - 2); i += 3) {
-            result += toBase64Table[data[i] >> 2];
-            result += toBase64Table[((data[i] & 0x03) << 4) + (data[i + 1] >> 4)];
-            result += toBase64Table[((data[i + 1] & 0x0f) << 2) + (data[i + 2] >> 6)];
-            result += toBase64Table[data[i + 2] & 0x3f];
+        for (let i = 0; i < (length - 2); i += 3) {
+            result += this.toBase64Table[data[i] >> 2];
+            result += this.toBase64Table[((data[i] & 0x03) << 4) + (data[i + 1] >> 4)];
+            result += this.toBase64Table[((data[i + 1] & 0x0f) << 2) + (data[i + 2] >> 6)];
+            result += this.toBase64Table[data[i + 2] & 0x3f];
```

Comment:
Why a bunch of `this.` all of a sudden?

Reply:
Before we did on top: `var toBase64Table = this.toBase64Table;`

Which is unnecessary. You can just call from this.

Reply:
The extra indirection might cause performance hiccups though. But we'll have to see during testing.